docs/content/sips/022-validate-target-environment.md

```diff
@@ -0,0 +1,75 @@
+title = "SIP 022 - Target Environment Validation"
+template = "main"
+date = "2025-08-07T00:00:00Z"
+---
+
+Summary: Provide a mechanism to validate that a Spin application is compatible with its intended deployment environment(s).
+
+Owner(s): till@tillschneidereit.net
+
+Created: Aug 7, 2025
+
+## Background
+
+When developing a Spin application, developers might use features that are only available in a specific Spin environment. For example, an application might use a custom trigger or a host capability that is only available in a specific deployment environment, or in a self-hosted Spin instance with specific plugins.
+
+Currently, there is no way to declare the intended deployment environment for an application and have Spin validate that the application is compatible with it. This can lead to situations where an application builds successfully but fails at runtime in the target environment due to missing dependencies, unsupported features, or other incompatibilities.
+
+This SIP proposes a new mechanism to address this problem by allowing developers to specify the target environments for their applications and have Spin validate the compatibility of the application with those environments at build time.
+
+## Proposal
+
+The proposal is to introduce a new `targets` field in the `[application]` section of the `spin.toml` manifest file. This field will contain a list of identifiers for the target environments. Spin will use these identifiers to fetch environment definition files and validate the application against them.
+
+### The `targets` field in `spin.toml`
+
+The `targets` field is a list of strings or tables that identify the target environments. It can be specified in the `[application]` section of `spin.toml` like this:
+
+```toml
+[application]
+name = "my-app"
+version = "1.0.0"
+targets = ["spin-up:3.3", "spinkube:0.4", { registry = "ghcr.io/my-org/environments", id = "my-env:1.0" }, { path = "local-env.toml" }]
+```
+
+Each entry in the list can be:
+- A string that identifies an environment in the default registry (e.g., `"spin-up:3.3"`).
+- A table with `registry` and `id` keys to identify an environment in a custom OCI registry.
```

Comment:
We should permit `wkg` registry names (bare domain) here with the `/.well-known/wasm-pkg/registry.json` indirection.

Reply:
Sure; I'm suggesting that we could reuse the domain -> OCI mapping that wkg already implements to allow nicer specifications of targets from registries.

Reply:
Ah sorry thanks for the clarification

Reply:
This feels like a convenience feature we could add without breaking back compat if the feature achieves traction?  Or are we sufficiently confident that other folks will be publishing environment TOMLs that we need to offer this in the first drop?